test/e2e/azure_test.go

```diff
@@ -937,6 +955,28 @@ var _ = Describe("Workload cluster creation", func() {
 		})
 	})
 
+	Context("Creating an AKS cluster with the ASO API [Managed Kubernetes][OPTIONAL]", func() {
```

Comment:
FYI I'm planning on a small test-infra change to the required AKS presumbit jobs to skip `[OPTIONAL]` tests since they currently run all of the `[Managed Kubernetes]` tests.

Reply:
Actually not: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4791#discussion_r1583614560